alchemist/alchemist-fxui/src/main/java/it/unibo/alchemist/boundary/gui/utility/FXResourceLoader.java

```diff
@@ -0,0 +1,109 @@
+package it.unibo.alchemist.boundary.gui.utility;
+
+import java.io.IOException;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Node;
+import javafx.scene.paint.Color;
+import jiconfont.IconCode;
+import jiconfont.icons.google_material_design_icons.GoogleMaterialDesignIcons;
+import jiconfont.javafx.IconFontFX;
+import jiconfont.javafx.IconNode;
+import org.kaikikm.threadresloader.ResourceLoader;
+
+/**
+ * Utility class for loading resources related to layout.
+ */
+public final class FXResourceLoader {
+    private static final String XML_RESOURCE_PATH = "it/unibo/alchemist/gui/view/";
+    private static final String LAYOUT_EXTENSION = ".fxml";
+    private static final String STYLE_EXTENSION = ".css";
+
+    /**
+     * Private, empty, constructor.
+     */
+    private FXResourceLoader() {
+        throw new AssertionError("Suppress default constructor for noninstantiability");
+    }
+
+    /**
+     * Loads a layout FXML file returning the base pane defined by the layout.
+     *
+     * @param <T>          the generic type of pane
+     * @param paneInstance the class of pain to load
+     * @param controller   the controller to associate to that layout
+     * @param layoutName   the name of the layout; it should be the file name without
+     *                     extension
+     * @return the pane defined by the layout
+     * @throws IOException if it cannot load the file for some reason
+     */
+    @SuppressWarnings("unchecked") // Passing a wrong class would be stupid
+    public static <T extends Node> T getLayout(final Class<T> paneInstance, final Object controller,
```

Comment:
It's unnecessary since the type of `T` is resolved through type inference, I see that initially the getLayout method only accepted a `Class<T> paneInstance` so it's probably just a leftover.
Removed.